src/deroff.rs

```diff
@@ -360,16 +360,33 @@ impl Deroffer {
         })
     }
 
-    fn skip_char<'a>(&self, s: &'a str, amount: Option<usize>) -> &'a str {
+    fn skip_char(&mut self, amount: usize) {
+        self.s = self.s.get(amount..).unwrap_or("").to_owned();
+    }
+
+    /* fn skip_char<'a>(&self, s: &'a str, amount: Option<usize>) -> &'a str {
         let amount = amount.unwrap_or(1);
         s.get(amount..).unwrap_or("")
+    } */
+
+    fn skip_leading_whitespace(&mut self) {
+        self.s = self.s.trim_start().to_owned();
```

Comment:
Same for this.

Reply:
Same for this one, in the python it just makes a call to `lstrip` which does the same as `trim_start`